apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/Preconditions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Linq;
+
+namespace Google.Cloud.Spanner
+{
+    internal static class Preconditions
```

Comment:
We already have GaxPreconditions - I'd suggest using those where possible. Happy to add methods to it if necessary.
(We originally had extension methods as well, but decided against them in the end.)